### DISCUSSION TITLE
Postgres instructions don't seem to be working for Mac M1 zsh

### DISCUSSION THREAD

**Discussion by geoidesic** [zotonic/zotonic#3331] [Q&A] [answered]
I notice a few instructions on the website:
E.g. here: https://zotonic.com/docs/1529/troubleshooting#ref-troubleshooting-installation
and here: https://zotonic.com/docs/1527/sites#guide-create-site

None of the instructions wrt Postgres seem to work on Mac OS 13.2.1 (zsh, M1 ARM architecture).

Specifically...
## 1. 
```
% sudo -u postgres psql
sudo: unknown user: postgres
sudo: error initializing audit plugin sudoers_audit
```

and...
## 2. 
```
% ls /etc/postgresql
ls: /etc/postgresql: No such file or directory
```

And yes, I have installed Postgres via Brew. 

## 3.
I am able to login to the management site using the password shown via `bin/zotonic config` – however, the instructions here: https://zotonic.com/docs/1528/the-status-site#ref-status-site, indicating that the password is stored in `~/.config/zotonic/config/[release]/zotonic.config` seems wrong:
```
% cat ~/.config/zotonic/config/[release]/zotonic.config
zsh: no matches found: ~/.config/zotonic/config/[release]/zotonic.config
% ls ~/.config/zotonic/config/[release]/zotonic.config
zsh: no matches found: ~/.config/zotonic/config/[release]/zotonic.config
```

Probably that bit should just be removed from the documentation because it will vary by platform and it should just say something like "the password will be stored in the zotonic config files, the location of which can be found using the instructions below".

## 4.
I tried adding a new site but got an error indicating that Postgres users are not setup correctly and due to first issue above I'm not able to amend that.

<img width="347" alt="Screenshot 2023-03-17 at 11 29 58" src="https://user-images.githubusercontent.com/4637995/225892914-cc7b95a4-2768-4184-a41e-0881b1d113cd.png">

**Comment by mworrell** [comment, marked as answer]
On macOS I like to use the Postgres.app  

https://postgresapp.com

Maybe we can add that one to the instructions.  It is much easier than using brew (etc.), especially when upgrading the database version.

**Reply by geoidesic** [reply]
That doesn't run on M1 architecture, except in Universal mode which is going to be an awful experience.

**Reply by geoidesic** [reply]
Actually scratch that... I'm thinking of Rosetta... Universal should be fine.

**Reply by geoidesic** [reply]
I had a go... it installs, but it's not clear how to work with it to make it compatible with the Zotonic config. By default, clicking "Initialise" creates this:
<img width="628" alt="Screenshot 2023-03-17 at 15 23 09" src="https://user-images.githubusercontent.com/4637995/225947795-e8f3bef7-89b8-4888-931d-c30777620da8.png">

So... how do I then create the Zotonic database and configure it so that it works with the Zotonic config?

**Reply by geoidesic** [reply]
By adding this to my `.zhsrc` 
```
export PATH="/Applications/Postgres.app/Contents/Versions/latest/bin:$PATH"
```
I then had access to `createdb zotonic`, which added a database called `zotonic`.

**Comment by geoidesic** [comment]
Which I then deleted using `dropdb zotonic`. 
Then I was able to follow the rest of the instructions here: https://zotonic.com/docs/1527/sites#guide-create-site
And that enabled the Manage Sites interface to be able to create a new site.

**Comment by mworrell** [comment]
This initial database setup is always tricky.  Not in the least because of all the variations in PostgreSQL installations.

Would be nice if we could have an easy step-by-step guide that everyone could follow.
